### ISSUE TITLE
Flakey tests

### ISSUE THREAD

**Issue by kgizdov** [wakatime/wakatime-cli#733]
**Expected behavior (what should have happened)**:
```bash
$ go test -tags=integration ./main_test.go
ok  	command-line-arguments	1.040s
```
**Actual behavior (what went wrong)**:
```bash
$ go test -tags=integration ./main_test.go
...
./build/wakatime-cli-linux-amd64 --log-file /tmp/TestSendHeartbeats_ExtraHeartbeats3249327417/002/1660507848 --api-url http://127.0.0.1:35321 --key 00000000-0000-4000-8000-000000000000 --config /tmp/TestSendHeartbeats_ExtraHeartbeats3249327417/001/wakatime.cfg244993821 --entity testdata/main.go --extra-heartbeats true --sync-offline-activity none --cursorpos 12 --offline-queue-file /tmp/TestSendHeartbeats_ExtraHeartbeats3249327417/001/2635607406 --lineno 42 --lines-in-file 100 --time 1585598059 --hide-branch-names .* --write --verbose
...
--- FAIL: TestSendHeartbeats_ExtraHeartbeats (0.12s)
    main_test.go:264:
        	Error Trace:	main_test.go:264
        	Error:      	Not equal:
        	           	expected: 2
        	           	actual  : 1
        	Test:       	TestSendHeartbeats_ExtraHeartbeats
...
FAIL
FAIL	command-line-arguments	1.026s
FAIL
```
**Environment**:
- OS: linux
- Platform: amd64
- Details: Arch Linux x86_64, Go 1.18.2, GCC 12.1.0

**Logs**:
[wakatime.log](https://github.com/wakatime/wakatime-cli/files/8744932/wakatime.log)

**Comment by gandarez** [comment]
Our pipeline runs integration tests for macOS, Windows and Linux and it hasn't been failing. Have you tried running more than once? Also try running `make test-integration`.

**Comment by alanhamlett** [comment]
`TestSendHeartbeats_ExtraHeartbeats` is one of the flakey tests that sometimes fails and we're not sure why.

**Comment by kgizdov** [comment]
I'm shipping this for Arch Linux and running your Makefile directly has some side-effect which I'd like to avoid. This is why I'm running it manually to reproduce the build and test configs for Arch. But I may be doing something wrong entirely. You can see the `build()` and `check()` functions which compile from source and then test the binary [here](https://github.com/archlinux/svntogit-community/blob/9b01d7e64a7e450867f9ad4f5d381c55803b4cc3/trunk/PKGBUILD#L22-L52):
```bash
build() {
  cd "${srcdir}/${pkgname}-cli"


  mkdir -p build  # create build dir


  export CGO_CPPFLAGS="${CPPFLAGS}"
  export CGO_CFLAGS="${CFLAGS}"
  export CGO_CXXFLAGS="${CXXFLAGS}"
  export CGO_LDFLAGS="${LDFLAGS}"


  local _date="$(date -u '+%Y-%m-%dT%H:%M:%S %Z')"
  local _commit="$(git rev-parse --short HEAD)"


  CGO_ENABLED="0" go build -o build/"${_binname}" \
    -trimpath \
    -buildmode=pie \
    -mod=readonly \
    -modcacherw \
    -ldflags "-linkmode external -extldflags '${LDFLAGS}' -X '${_repo}/pkg/version.OS=linux' -X '${_repo}/pkg/version.Arch=amd64' -X '${_repo}/pkg/version.BuildDate=${_date}' -X '${_repo}/pkg/version.Commit=${_commit}' -X '${_repo}/pkg/version.Version=${pkgver}'"
}


check () {
  cd "${srcdir}/${pkgname}-cli"
  echo "wakatime-cli" > ./.wakatime-project
  # some weird test assumptions fail
  sed \
    -e "s/<local-build>/${pkgver}/" \
    -e "s/Equal(t, 2, offlineCount)/Equal(t, 1, offlineCount)/" \
    -i ./main_test.go
  CGO_ENABLED="0" go test -tags=integration ./main_test.go
}
```

Something that has always been a problem for the test is that for some reason it always thinks it's a local build and seems to ignore the setting of the version number and also the heartbeat count. Any tips or help will be appreciated. Thanks.

**Comment by kgizdov** [comment]
If it helps you, I can say that for version `1.42.2` the heartbeat test worked correctly.

**Comment by alanhamlett** [comment]
Fixed with #743.
